files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js

```diff
@@ -676,78 +676,81 @@ class GroupedWindowListApplet extends Applet.Applet {
     }
 
     handleScroll(e, sourceFromAppGroup) {
-        if ((this.state.settings.scrollBehavior === 1 && this.state.settings.leftClickAction !== 3)
-            || (e && sourceFromAppGroup && !this.state.settings.thumbnailScrollBehavior)) {
-            return;
-        }
-
-        this.state.set({scrollActive: true});
-
-        let isAppScroll = this.state.settings.scrollBehavior === 2;
-        let direction, source;
-
-        if (sourceFromAppGroup) {
-            isAppScroll = false;
-            direction = e ? e.get_scroll_direction() : 1;
-            source = sourceFromAppGroup;
-        } else {
-            direction = e.get_scroll_direction();
-            source = e.get_source()._delegate;
-        }
-        let lastFocusedApp, z, count
-
-        if (isAppScroll) {
-            lastFocusedApp = this.appLists[this.state.currentWs].listState.lastFocusedApp;
-            if (!lastFocusedApp) {
-                lastFocusedApp = this.appLists[this.state.currentWs].appList[0].groupState.appId
-            }
-            let focusedIndex = findIndex(this.appLists[this.state.currentWs].appList, function(appGroup) {
-                return appGroup.groupState.metaWindows.length > 0 && appGroup.groupState.appId === lastFocusedApp;
-            });
-            z = direction === 0 ? focusedIndex - 1 : focusedIndex + 1;
-            count = this.appLists[this.state.currentWs].appList.length - 1;
-        } else {
-            if (!source.groupState || source.groupState.metaWindows.length < 1) {
-                return;
+        if( (this.state.settings.thumbnailScrollBehavior) || 
+            (this.state.settings.leftClickAction === 3 && this.state.settings.scrollBehavior !== 3
+             && !e && sourceFromAppGroup)  ||
+            (this.state.settings.leftClickAction !== 3 && this.state.settings.scrollBehavior === 3               
+                        && e && !sourceFromAppGroup)  || 
+            (this.state.settings.leftClickAction === 3 && this.state.settings.scrollBehavior === 3)) {
+
+            this.state.set({scrollactive: true});
+
+            let isappscroll = this.state.settings.scrollbehavior === 2;
+            let direction, source;
+
+            if (sourcefromappgroup) {
+                isappscroll = false;
+                direction = e ? e.get_scroll_direction() : 1;
+                source = sourcefromappgroup;
```

Comment:
same

Reply:
That along with the rest should be squared up now.